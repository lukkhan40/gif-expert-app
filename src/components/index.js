export * from "./AddCategory";
export * from "./GifGrid/GifGrid";
export * from "./GifGrid/GifGridItem";
