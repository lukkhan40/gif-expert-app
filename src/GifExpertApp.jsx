import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);

  const addCategory = (category) => {
    if (categories.includes(category)) return;

    setCategories([category, ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory addCategory={addCategory} />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};
