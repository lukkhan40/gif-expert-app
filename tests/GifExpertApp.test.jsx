import { render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe("Prueba <GifExpertApp/>", () => {
  test("Evaluar el snapshot", () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });
});
