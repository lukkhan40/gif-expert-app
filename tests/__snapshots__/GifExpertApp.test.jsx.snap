// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba <GifExpertApp/> Evaluar el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      name="input"
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <ol />
</div>
`;
